src/PathautoManager.php

```diff
@@ -122,15 +128,31 @@ public function __construct(ConfigFactoryInterface $config_factory, ModuleHandle
   /**
    * {@inheritdoc}
    */
-  public function createAlias($module, $op, $source, $data, $type = NULL, $langcode = LanguageInterface::LANGCODE_NOT_SPECIFIED) {
-    $config = $this->configFactory->get('pathauto.settings');
-
+  public function createAlias(EntityInterface $entity, $op) {
     // Retrieve and apply the pattern for this content type.
-    $pattern = $this->getPatternByEntity($module, $type, $langcode);
+    $pattern = $this->getPatternByEntity($entity);
+    if (empty($pattern)) {
+      // No pattern? Do nothing (otherwise we may blow away existing aliases...)
+      return NULL;
+    }
+
+    $entity_type_id = $entity->getEntityTypeId();
+    $source = $entity->urlInfo()->toString();
```

Comment:
wondering if it would be easier to just inline some of those variables.. this can also be written as $entity->url() btw.